doc/stdenv/meta.chapter.md

```diff
@@ -125,17 +125,38 @@ $ cd path/to/nixpkgs
 $ nix-build -A your-package.tests
 ```
 
+Note that Hydra and [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review) don't build these derivations by default, but they are useful pointers for Nixpkgs' CI ([`@ofborg`](https://github.com/NixOS/ofborg)) that may help identify issues when changes such as version bumps are directly made to the package.
+
 #### Package tests {#var-meta-tests-packages}
 
-Tests that are part of the source package are often executed in the `installCheckPhase`.
+Tests that are part of the source package are often executed in the `installCheckPhase`. This phase is also suitable for performing a `--version` test for packages that support such flag. Here's an example:
+
+```nix
+# Say the package is git
+stdenv.mkDerivation(finalAttrs: {
```

Comment:
this style seems less common than `mkDerivation rec {`, is there a reason we'd want to show `finalAttrs` here?

Reply:
> this style seems less common than `mkDerivation rec {`, is there a reason we'd want to show `finalAttrs` here?

using `finalAttrs` is the modern way to `mkDerivation` nowadays, as it makes overriding the package much easier when the package references it self, and especially if you want to make it easy for the user reading this to maybe add a `passthru.tests.comprehensive` that will use `finalAttrs.finalPackage` which is not possible with the old `mkDerivation rec {` pattern.

Reply:
> using `finalAttrs` is the modern way to `mkDerivation` nowadays

Do you have a reference for that? https://nixos.org/manual/nixpkgs/unstable/#sec-using-stdenv is still showing `rec`

> it makes overriding the package much easier when the package references it self, and especially if you want to make it easy for the user reading this to maybe add a `passthru.tests.comprehensive` that will use `finalAttrs.finalPackage` which is not possible with the old `mkDerivation rec {` pattern

TIL, very cool

Reply:
Some more ongoing discussion on this in #315337